pkg/services/ngalert/notifier/channels/wecom.go

```diff
@@ -14,65 +14,68 @@ import (
 	"github.com/grafana/grafana/pkg/services/notifications"
 )
 
-type WeComConfig struct {
-	*NotificationChannelConfig
-	URL     string
-	Message string
-	Title   string
+type wecomSettings struct {
+	URL     string `json:"url" yaml:"url"`
+	Message string `json:"message,omitempty" yaml:"message,omitempty"`
+	Title   string `json:"title,omitempty" yaml:"title,omitempty"`
+}
+
+func buildWecomSettings(factoryConfig FactoryConfig) (wecomSettings, error) {
+	var settings = wecomSettings{
```

Comment:
I thought about this strategy for Slack, but realized that it has very slightly different behavior than using an if statement to fill the default directly:

Assume the settings in the request are:
```
{
    "myField": ""
}
```
This yields:
```
var settings = blahSettings{
    MyField: DefaultValue
}

unmarshalSettings(&settings)

// gives settings.MyField == ""
```

Where this yields:
```
var settings blahSettings
unmarshalSettings(&settings)

if settings.MyField == "" {
    settings.MyField = DefaultValue
}

// gives settings.MyField == DefaultValue
```

Reply:
applied the fix you suggested!